workflow_plandefinition/models/workflow_plan_definition_action.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import fields, models, api, exceptions
+
+
+class PlanDefinitionAction(models.Model):
+    _name = 'workflow.plan.definition.action'
+    _description = 'Medical Plan Definition Action'
+    _inherit = 'mail.thread'
+    _parent_name = 'parent_id'
+    _parent_store = True
+    _parent_order = 'name'
+    _order = 'parent_left'
+    _rec_name = 'complete_name'
+    _order = 'id asc'
+
+    @api.one
+    @api.depends('name', 'parent_id')
+    def _compute_complete_name(self):
+        """ Forms complete name of action from parent to child action. """
+        name = self.name
+        current = self
+        while current.parent_id:
```

Comment:
Please add cycle detection on the `parent_id` to ensure no infinite loop. Exmaple: https://github.com/LasLabs/vertical-medical/blob/3abb2cfe587143dd3b240f710bac7acaafff7234/medical_pathology/models/medical_pathology_category.py#L41

Reply:
done!